xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-test/xwiki-platform-administration-test-docker/src/test/it/org/xwiki/administration/test/ui/RegisterIT.java

```diff
@@ -0,0 +1,301 @@
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.administration.test.ui;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebElement;
+import org.xwiki.administration.test.po.RegistrationModal;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+import org.xwiki.test.ui.po.AbstractRegistrationPage;
+import org.xwiki.test.ui.po.RegistrationPage;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test the user registration feature.
+ * <p>
+ * The tests in this class are parametrized with the values:
+ * <ul>
+ *   <li><b>useLiveValidation:</b> when {@code true} activates the client side validation of the registration form</li>
+ *   <li>
+ *     <b>isModal:</b> when {@code true} the user creation modal from the administration is used, otherwise the guest
+ *     user registration form is used
+ *   </li>
+ * </ul>
+ * <p>
+ * Three combinations of these parameters are tested:
+ * <ul>
+ *   <li>useLiveValidation + !isModal</li>
+ *   <li>!useLiveValidation + !isModal</li>
+ *   <li>useLiveValidation + isModal</li>
+ * </ul>
+ *
+ * @version $Id$
+ * @since 13.4RC1
+ * @since 12.10.7
+ */
+@UITest
+class RegisterIT
+{
+    /**
+     * Returns a stream of combinations of parameters to test. The first value is {@code useLiveValidation} and the
+     * second {@code isModal}.
+     *
+     * @return the tested combination of {@code useLiveValidation} and {@code isModal}
+     */
+    private static Stream<Arguments> testsParameters()
+    {
+        return Stream.of(
+            Arguments.of(true, false),
+            Arguments.of(false, false),
+            Arguments.of(true, true)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("testsParameters")
+    @Order(1)
+    void registerJohnSmith(boolean useLiveValidation, boolean isModal, TestUtils testUtils) throws Exception
+    {
+        AbstractRegistrationPage registrationPage = setUp(testUtils, useLiveValidation, isModal);
+        assertTrue(validateAndRegister(testUtils, useLiveValidation, isModal, registrationPage));
```

Comment:
better use assertThat with a message. If I'm correct that's now supposed to be our best practice

Reply:
> We should use Hamcrest assertThat() method when it makes sense (when there's no good JUnit 5.x assertion alternative or without a good reporting when the assertion fails. For example for asserting that a list contains a given entry).

Here `assertTrue` is imo perfectly fine, but it wouldn't hurt to have a failure message indeed.